src/main/kotlin/de/cyface/collector/handler/AuthorizationHandler.kt

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2022 Cyface GmbH
+ *
+ * This file is part of the Cyface Data Collector.
+ *
+ * The Cyface Data Collector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Cyface Data Collector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface Data Collector. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.collector.handler
+
+import de.cyface.api.Authorizer
+import de.cyface.api.DatabaseConstants
+import de.cyface.api.PauseAndResumeStrategy
+import de.cyface.api.UserRetriever
+import de.cyface.api.model.Role
+import de.cyface.api.model.User
+import io.vertx.core.Future
+import io.vertx.core.Handler
+import io.vertx.core.Promise
+import io.vertx.core.http.HttpServerRequest
+import io.vertx.core.json.JsonObject
+import io.vertx.ext.auth.mongo.MongoAuthentication
+import io.vertx.ext.auth.mongo.MongoAuthorization
+import io.vertx.ext.mongo.MongoClient
+import io.vertx.ext.web.RoutingContext
+import org.apache.commons.lang3.Validate
+import org.slf4j.LoggerFactory
+import java.util.Locale
+import java.util.stream.Collectors
+
+/**
+ * A `RequestHandler` to ensure correct user authentication.
+ * This should be one of the first handlers on each route requiring authorization.
+ *
+ * @author Klemens Muthmann
+ * @version 1.0.0
+ * @property authProvider An auth provider used by this server to authenticate against the Mongo user database.
+ * @property mongoDatabase The database to check which users a {@code manager} user has access to.
+ * @property strategy The pause and resume strategy to be used which wraps async calls
+ *                    in {@link #handle(RoutingContext)}.
+ */
+class AuthorizationHandler(
```

Comment:
Do we really want to have this code duplicated in collector, provider, etc.? This would mean that we would also need to duplicate all code in the project `api` to both repositories (or why the distinction?) and apply all changes to both, in the future. I don't see a reason why we should do this.

I.e. shouldn't we just replace the `api/Authorizer` with this `AuthorizationHandler` so that we can use the refactoring in the provider project, too?

Reply:
Maybe. But in the spirit of micro services I wanted to keep the change local, until we know this really works. Even then duplication must not be a bad thing. We need to weight independence of individual services vs effort to fix problems or extend duplicate code.